### PR TITLE
SDP-370 Filter assets by wallets

### PR DESCRIPTION
**Pull request by marwen-abid**
### What
Add a new parameter to `GET /assets` to filter assets by wallet id. 
e.g. `GET/assets?wallet=360383c7-9e6a-4713-8293-9c3565877a0f` 

### Why
We added association between wallets and their supported assets. 

### Known limitations

[TODO or N/A]

#### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR does not mix refactoring changes with feature changes (split into two PRs otherwise).
* [ ] This PR's title starts with the name of the package, area, or subject affected by the change.

#### Thoroughness

* [ ] This PR adds tests for the new functionality or fixes.
* [ ] This PR contains the link to the Jira ticket it addresses.

#### Configs and Secrets

* [ ] No new CONFIG variables are required -OR- the new required ones were added to the helmchart's [`values.yaml`] file.
* [ ] No new CONFIG variables are required -OR- the new required ones were added to the deployments ([`pr-preview`], [`dev`], [`demo`], `prd`).
* [ ] No new SECRETS variables are required -OR- the new required ones were mentioned in the helmchart's [`values.yaml`] file.
* [ ] No new SECRETS variables are required -OR- the new required ones were added to the deployments ([`pr-preview secrets`], [`dev secrets`], [`demo secrets`], `prd secrets`).

#### Release

* [ ] This is not a breaking change.
* [ ] **This is ready for production.**. If your PR is not ready for production, please consider opening additional complementary PRs using this one as the base. Only merge this into `develop` or `main` after it's ready for production!

#### Deployment

* [ ] Does the deployment work after merging?

[`values.yaml`]: ../helmchart/sdp/values.yaml
[`pr-preview`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/stellar-disbursement-platform/backend-helm-values
[`dev`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/backend-helm-values
[`demo`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-backend-helm-values
[`pr-preview secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/externalsecrets-common-previews.yaml#L241-L346
[`dev secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/stellar-disbursement-platform-externalsecrets.yaml
[`demo secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-sdp-externalsecrets.yaml

### CODE REVIEW

File: internal/services/wallets/wallets_testnet.go [hunk 1]
```diff
@@ -19,4 +19,16 @@ var TestnetWallets = []data.Wallet{
 			},
 		},
 	},
+	{
+		Name:              "Vibrant Assist",
+		Homepage:          "https://vibrantapp.com",
```
Comment (marcelosalloum) [inline internal/services/wallets/wallets_testnet.go:24]: Doesn't aid has a different website?

File: internal/data/assets.go [hunk 2]
```diff
@@ -77,20 +102,41 @@ func (a *AssetModel) GetByCodeAndIssuer(ctx context.Context, code, issuer string
...
 	query := `
 		SELECT 
-			a.id, 
-			a.code, 
-			a.issuer,
-			a.created_at,
-			a.updated_at,
-			a.deleted_at
+			a.*
 		FROM
```
Comment (marcelosalloum) [inline internal/data/assets.go:136]: 🎉

File: internal/data/assets.go [hunk 3]
```diff
@@ -77,20 +102,41 @@ func (a *AssetModel) GetByCodeAndIssuer(ctx context.Context, code, issuer string
...
+	query := `
+		SELECT 
+			a.*
+		FROM 
+			assets a
+		JOIN 
+			wallets_assets wa ON a.id = wa.asset_id
+		WHERE 
+		    deleted_at IS NULL
+			AND wa.wallet_id = $1
```
Comment (marcelosalloum) [inline internal/data/assets.go:117]: The indentation here is a bit odd.

File: internal/data/assets.go [hunk 4]
```diff
@@ -24,6 +24,31 @@ type AssetModel struct {
...
+	err := a.dbConnectionPool.GetContext(ctx, &asset, query, assetID, walletID)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return nil, ErrRecordNotFound
+		}
+		return nil, fmt.Errorf("querying asset ID %s and wallet ID %s: %w", assetID, walletID, err)
+	}
+	return &asset, nil
+}
+
```
Comment (marcelosalloum) [inline internal/data/assets.go:51]: Is this method needed? Can we delete it?

**Review by marcelosalloum — APPROVED** [review]
LGTM! Plz feel free to merge it after addressing the comments below.

File: internal/services/wallets/wallets_testnet.go (hunk 1 above)
Comment (marwen-abid) [reply internal/services/wallets/wallets_testnet.go]: fixed in c1f64ec12d0df4c2cc3f0e8b3d01af2b92260d98

File: internal/data/assets.go (hunk 3 above)
Comment (marwen-abid) [reply internal/data/assets.go]: fixed in c3b11e00d64721769e98ca0c497ab056b78e1139

**Pull request merged**